### ISSUE TITLE
[QUESTION] Can you use inheritance with provider generation blocks

### ISSUE THREAD

**Issue by itarchmerc** [gruntwork-io/terragrunt#1135]
I am using a terragrunt.hcl file at the root of my project to define my AWS provider as every sub-folder currently shares the same settings.  How do I extend the provider definition at a lower level?  The current issue is that I need to set ignore_tag_prefixes in the provider block for my vpc folder so tags created by EKS aren't overridden.  Right now I just recreated the entire provider block in that folder, but I feel like there is be a better way to extend provider blocks.  I wasn't sure if creating an alias was a better approach, but even then, it's mostly repetitive.  My folder structure is:

```
/project
  terragrunt.hcl
  /dev
  /stage
  /prod
  /tools
    /eks
      terragrunt.hcl
    /vpc
      terragrunt.hcl. <= Need to add ignore_tag_prefixes here
```

Thanks for this project!  It has helped solve many of my issues with standard terraform.

**Comment by yorinasub17** [comment]
The only way to accomplish this currently is to use `read_terragrunt_config` with `find_in_terraform_folders` so that you can reference variables in your folder structure in the parent config.

On a separate note, in general, this is an anti pattern (having customizations at the root) as it makes it hard to follow your terragrunt configuration since you are using the root configurations as a function, but we don't have a good solution currently to support use cases that depend on function-like usage.

We have an RFC for a new block that is intended to solve this, but it is still in development. You can see the RFC here: https://github.com/gruntwork-io/terragrunt/blob/master/_docs/rfc/imports.md

**Comment by yorinasub17** [comment]
Closing this as stale. If this is still an issue, please open a new ticket with updated context. Thanks!

PS: I think there is a potential way to implement this now using a combination of `json` terraform syntax, multiple includes, and `generate` blocks. Happy to provide an example if you can provide a concrete use case, where you can share the common provider config, and what you want to override.
